Realm/RealmSwift/Realm.swift

```diff
@@ -21,27 +21,135 @@ import Realm.Private
 
 // MARK: Object Retrieval
 
-public func objects<T: Object>(type: T.Type) -> Results<T> {
-    return Results<T>(RLMGetObjects(RLMRealm.defaultRealm(), T.className(), nil))
+/**
+Returns all objects of the given type in the specified Realm default realm,
+or in the default realm if the `realm` argument is omitted.
+
+:param: type  The type of the objects to be returned.
+:param: realm The Realm instance to query.
+              The default Realm will be used if this argument is omitted.
+
+:returns: Results with all objects of the given type in the specified Realm.
+*/
+public func objects<T: Object>(type: T.Type, inRealm realm: Realm = defaultRealm()) -> Results<T> {
```

Comment:
This is one of the few non-doc, non-trivial changes that I kept in this PR because it goes in concert with  the docs for it.